### PR TITLE
XCM Overhaul

### PR DESCRIPTION
**Pull request by eshaben**
### Description/Original PRs

Original PR: https://github.com/moonbeam-foundation/moonbeam-docs/pull/777

### Checklist

- [ ] If this requires removing old images from the `moonbeam-docs` repo, I have created a corresponding PR
- [ ] If this requires adding/updating/deleting redirects, I have created a corresponding PR in the `moonbeam-mkdocs` repo
- [ ] If this requires removing old variables from the `moonbeam-docs` repo, I have created a corresponding PR

### Corresponding PRs

Please link to any additional corresponding PRs here.

### CODE REVIEW

**Comment by eshaben** [comment]
@wuzhong-papermoon I've added all but one translated page (still waiting on the final review process of the X-Tokens Precompile page - it's in the queue on Trello). There are also some minor changes I committed in English, can you please review those changes and translate them? Let me know if you have any questions 🙂

**Review by wuzhong-papermoon — COMMENTED** [review]
Please have a look and let me know if it works.

File: builders/interoperability/xcm/core-concepts/.pages [hunk 1]
```diff
@@ -0,0 +1,6 @@
+title: Core Concepts
```
Comment (eshaben) [inline builders/interoperability/xcm/core-concepts/.pages:1]: @wuzhong-papermoon

File: builders/interoperability/xcm/remote-execution/.pages [hunk 2]
```diff
@@ -0,0 +1,7 @@
+title: Remote Execution
+nav:
+  - index.md
+  - 'Overview' : 'overview.md'
+  - 'Computed Origin Accounts': 'computed-origins.md'
```
Comment (eshaben) [inline builders/interoperability/xcm/remote-execution/.pages:5]: @wuzhong-papermoon

File: builders/interoperability/xcm/remote-execution/index.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+title: Cross-Chain Remote Execution
+description: Learn about the basics of remote execution and how to perform remote cross-chain execution on other chains in the ecosystem and remote EVM calls on Moonbeam.
```
Comment (eshaben) [inline builders/interoperability/xcm/remote-execution/index.md:3]: @wuzhong-papermoon

File: builders/interoperability/xcm/remote-execution/remote-evm-calls.md [hunk 4]
```diff
@@ -22,29 +22,26 @@ Moonbeam的EVM仅能通过[Ethereum Pallet](https://github.com/paritytech/fronti
...
+- [`BuyExecution`](/builders/interoperability/xcm/core-concepts/instructions#buy-execution){target=\_blank}
+- [`Transact`](/builders/interoperability/xcm/core-concepts/instructions#transact){target=\_blank}
+
+XCM execution happens through a [Computed Origin account mechanism](/builders/interoperability/xcm/remote-execution/computed-origins){target=\_blank}, which by default uses the source chain's Sovereign account in the destination chain. If `DescendOrigin` is included, Moonbeam will mutate the origin of the XCM call to a keyless account that a user from the source chain can control remotely via XCM. The remote EVM call is dispatched from that keyless account (or a related [proxy](/tokens/manage/proxy-accounts/){target=\_blank}). Therefore, because the transaction is not signed, it does not have the real `v-r-s` values of the signature, but `0x1` instead.
 
 第一条指令`DescendOrigin`将通过[XCM相关定义部分](#general-xcm-definitions)中描述的**Multilocation衍生账户**机制在Moonbeam端转移XCM调用的源头。远程EVM调用是从该无密钥帐户（或相关的[代理](/tokens/manage/proxy-accounts/){target=\_blank}）派遣的。因此，由于交易并没有签署，它没有真正签名的`v-r-s`值，反而会是`0x1`。
```
Comment (eshaben) [inline builders/interoperability/xcm/remote-execution/remote-evm-calls.md:44]: Looks like these sections aren't 1:1. Can you please provide the translation of the content on line 44? @wuzhong-papermoon

File: builders/interoperability/xcm/remote-execution/remote-evm-calls.md [hunk 5]
```diff
@@ -71,110 +68,76 @@ XCM队列的配置表明XCM消息应该设置为`20,000,000,000`权重单位（
...
+            - `gasLimit` - the gas limit for the Ethereum transaction
+            - `action` - the action to be executed, which provides two options: `Call` and `Create`. The current implementation of the [Ethereum XCM Pallet](https://github.com/moonbeam-foundation/moonbeam/tree/master/pallets/ethereum-xcm){target=\_blank} does not support the `CREATE` operation. Therefore, you can't deploy a smart contract through remote EVM calls. For `Call`, you'll need to specify the contract address you're interacting with
+            - `value` - the amount of native tokens to send
+            - `input` - the encoded call data of the contract interaction
 
-!!! 注意事项
-    当在使用`transactThroughProxy`时，只要此帐户已将**multilocation衍生账户**设置为您正在使用的基于Moonbeam的网络中类型为`any`的代理，EVM调用将根据您提供的**transactAs**帐户调度，并作为`msg.sender`。但是，交易费用仍由**multilocation衍生账户**支付，因此您需要确保它有足够的资金来支付这些费用。
-
-### 查看先决条件 {: #ethereumxcm-check-prerequisites}
+    === "Polkadot.js API Example"
```
Comment (eshaben) [inline builders/interoperability/xcm/remote-execution/remote-evm-calls.md]: @wuzhong-papermoon

File: builders/interoperability/xcm/remote-execution/remote-evm-calls.md [hunk 6]
```diff
@@ -71,110 +68,76 @@ XCM队列的配置表明XCM消息应该设置为`20,000,000,000`权重单位（
...
-    X1: {
-      AccountId32: {
-        network: { westend: null },
-        id: decodedAddress,
-      },
-    },
-  },
-}
-```
+    === "Polkadot.js API Example"
```
Comment (eshaben) [inline builders/interoperability/xcm/remote-execution/remote-evm-calls.md:99]: @wuzhong-papermoon

File: builders/interoperability/xcm/remote-execution/remote-evm-calls.md [hunk 7]
```diff
@@ -71,110 +68,76 @@ XCM队列的配置表明XCM消息应该设置为`20,000,000,000`权重单位（
...
 
---8<-- 'text/builders/interoperability/xcm/calculate-multilocation-derivative-account.md'
+The process for building and performing the remote execution can be summarized as follows:
 
-以Alice的中继链账户`5DV1dYwnQ27gKCKwhikaw1rz1bYdvZZUuFkuduB4hEK3FgDT`为例，您可以通过运行以下命令来计算他的Moonbase Alpha **multilocation衍生账户**：
+1. Calculate the call data for the EVM call that will be performed on Moonbase Alpha
+2. Use the EVM call data to generate the call data for the `transact` extrinsic of the Ethereum XCM Pallet on Moonbase Alpha
+3. Build the XCM message on the relay chain, which will include the `WithdrawAsset`, `BuyExecution`, and `Transact` instructions. In the `Transact` instruction, you'll use the Ethereum XCM `transact` call data
+4. Using Alice's account on the relay chain, you'll send the XCM message via the `send` extrinsic of the XCM Pallet
+5. Alice's Computed Origin account on Moonbase Alpha will dispatch the EVM call data
```
Comment (eshaben) [inline builders/interoperability/xcm/remote-execution/remote-evm-calls.md:118]: @wuzhong-papermoon

File: builders/interoperability/xcm/remote-execution/remote-evm-calls.md [hunk 8]
```diff
@@ -71,110 +68,76 @@ XCM队列的配置表明XCM消息应该设置为`20,000,000,000`权重单位（
...
+- 为**Computed Origin账户**提供资金，您可以按照[下一部分](#calculate-multilocation-derivative){target=\_blank}中的步骤获得该账户。该账户必须有足够的DEV Token（或Moonbeam/Moonriver网络中的GLMR/MOVR）来支付远程EVM调用的XCM执行成本。请注意，此衍生账户是将发送远程EVM调用的帐户（`msg.sender`）。因此，帐户必须满足正确执行EVM调用所需的任何条件。例如，如果您正在执行ERC-20转账，请确保拥有任何相关的ERC-20 Token
+- The address of your Computed Origin account. Please refer to the [Computed Origin](/builders/interoperability/xcm/remote-execution/computed-origins){target=\_blank} guide to learn how to calculate your Computed Origin address
+- To fund your Computed Origin account. The account must have enough DEV tokens (or GLMR/MOVR for Moonbeam/Moonriver) to cover the cost of the XCM execution of the remote EVM call. Note that this is the account from which the remote EVM call will be dispatched (the `msg.sender`). Consequently, the account must satisfy whatever conditions are required for the EVM call to be executed correctly. For example, hold any relevant ERC-20 token if you are doing an ERC-20 transfer
```
Comment (eshaben) [inline builders/interoperability/xcm/remote-execution/remote-evm-calls.md:127]: @wuzhong-papermoon

File: builders/interoperability/xcm/remote-execution/remote-evm-calls.md [hunk 9]
```diff
@@ -221,17 +184,30 @@ const xcmTransaction = {
...
+
 要更动编码调用数据的脚本，您需要添加逻辑为传送者（此例中为Alice）创建一个Keyring。接着您可以简单的使用`tx`并调用`paymentInfo`函数和输入Alice的Keyring。
 
+???+ code "Modified script"
+
+    ```js
+    --8<-- 'code/builders/interoperability/xcm/remote-execution/remote-evm-calls/estimate-required-weight.js'
+    ```
+
+The script, at the time of writing, returns an estimate of `3900000000` for `refTime` and `38750` for `proofSize`.
```
Comment (eshaben) [inline builders/interoperability/xcm/remote-execution/remote-evm-calls.md:199]: @wuzhong-papermoon

File: builders/interoperability/xcm/remote-execution/remote-evm-calls.md [hunk 10]
```diff
@@ -221,17 +184,30 @@ const xcmTransaction = {
...
+
 ### 为远程XCM执行构建XCM {: #build-xcm-remote-evm}
 
-在本示例中，您将构建一条XCM消息，通过[`Transact`](https://github.com/paritytech/xcm-format#transact){target=\_blank} XCM指令和[Ethereum XCM Pallet](https://github.com/moonbeam-foundation/moonbeam/tree/master/pallets/ethereum-xcm){target=\_blank}的`transact`函数从其中继链在Moonbase Alpha种执行远程EVM调用。
+Now that you've generated the call data for the EVM call, you're going to use the XCM Pallet on the relay chain to perform the remote execution. To do so, you'll use the `send` function, which accepts two parameters:
+
+- `dest` - the XCM versioned multilocation representing a chain in the ecosystem where the XCM message is being sent to (the target chain)
+- `message` - the SCALE-encoded versioned XCM message to be executed
```
Comment (eshaben) [inline builders/interoperability/xcm/remote-execution/remote-evm-calls.md]: @wuzhong-papermoon

File: builders/interoperability/xcm/remote-execution/remote-evm-calls.md [hunk 11]
```diff
@@ -276,25 +252,7 @@ const xcmTransaction = {
...
-
-        ??? code "完整脚本"
-
-            ```js
-            --8<-- 'code/builders/interoperability/xcm/remote-execution/remote-evm-calls/estimate-required-weight.js'
-            ```
-
-        截至撰写本脚本时，`refTime`和`proofSize`会分别返回`3900000000`和`38750`的预估数值
-
+    - The required weight for the transaction, which you calculated in the [Estimate Weight Required at Most](#estimate-weight-required-at-most) section
```
Comment (eshaben) [inline builders/interoperability/xcm/remote-execution/remote-evm-calls.md:253]: @wuzhong-papermoon

File: builders/interoperability/xcm/remote-execution/substrate-calls/.pages [hunk 12]
```diff
@@ -0,0 +1,5 @@
+title: Remote Substrate Calls
+nav:
+  - index.md
+  - 'Using the Substrate API': 'xcm-transactor-pallet.md'
+  - 'Using the Ethereum API': 'xcm-transactor-precompile.md'
```
Comment (eshaben) [inline builders/interoperability/xcm/remote-execution/substrate-calls/.pages:5]: @wuzhong-papermoon

File: builders/interoperability/xcm/remote-execution/substrate-calls/index.md [hunk 13]
```diff
@@ -0,0 +1,8 @@
+---
+title: Remote Substrate Calls
+description: Discover how to use the XCM Transactor Pallet and XCM Transactor Precompile to remotely call Substrate pallets on other chains in the ecosystem from Moonbeam.
+template: subsection-index-page.html
+hide: 
+ - toc
+ - feedback
+---
```
Comment (eshaben) [inline builders/interoperability/xcm/remote-execution/substrate-calls/index.md:8]: @wuzhong-papermoon

File: builders/interoperability/xcm/xc20/send-xc20s/.pages [hunk 14]
```diff
@@ -0,0 +1,6 @@
+title: XC-20转账
+nav:
+  - index.md
+  - 'Overview': 'overview.md'
+  - 'Using the Substrate API': 'xtokens-pallet.md'
+  - 'Using the Ethereum API': 'xtokens-precompile.md'
```
Comment (eshaben) [inline builders/interoperability/xcm/xc20/send-xc20s/.pages:6]: @wuzhong-papermoon

File: builders/interoperability/xcm/xc20/send-xc20s/index.md [hunk 15]
```diff
@@ -0,0 +1,8 @@
+---
+title: Send XC-20s via XCM
+description: This section shows you how to send XC-20s cross-chain from Moonbeam to other chains in the ecosystem using the X-Tokens Pallet and X-Tokens Precompile.
```
Comment (eshaben) [inline builders/interoperability/xcm/xc20/send-xc20s/index.md:3]: @wuzhong-papermoon

File: builders/interoperability/xcm/xcm-utils.md [hunk 16]
```diff
@@ -41,17 +41,21 @@ XCM Utilities预编译合约为开发者提供了直接在EVM中与XCM相关的
 
 此接口包含以下函数：
 
- - **multilocationToAddress**(*Multilocation memory* multilocation) — 只读函数，为给定multilocation返回multilocation的衍生账户
+ - **multilocationToAddress**(*Multilocation memory* multilocation) — 只读函数，为给定multilocation返回Computed Origin账户
  - **weightMessage**(*bytes memory* message) — 只读函数，返回XCM消息将在链上消耗的权重。消息参数必须为SCALE编码的XCM版本化的XCM消息
  - **getUnitsPerSecond**(*Multilocation memory* multilocation) — 只读函数，为以`Multilocation`形式给定的资产获取每秒单位数。multilocation必须描述一个可以支持作为费用支付的资产，例如[外部XC-20](/builders/interoperability/xcm/xc20/overview/#external-xc20s){target=\_blank}，否则此函数将回退当前调用(revert)
+ - **xcmExecute**(*bytes memory* message, *uint64* maxWeight) - **available on Moonbase Alpha only** -  executes a custom XCM message given the SCALE encoded versioned message to be executed and the maximum weight to be consumed. This function *cannot* be called from a smart contract due to the nature of the `Transact` instruction
+ - **xcmSend**(*Multilocation memory* dest, *bytes memory* message) - **available on Moonbase Alpha only** - sends a custom XCM message given the multilocation of the destination chain to send the message to and the SCALE encoded versioned message to be sent
```
Comment (eshaben) [inline builders/interoperability/xcm/xcm-utils.md:48]: @wuzhong-papermoon

File: builders/interoperability/xcm/xcm-utils.md [hunk 17]
```diff
@@ -41,17 +41,21 @@ XCM Utilities预编译合约为开发者提供了直接在EVM中与XCM相关的
...
 
-在XCM Utilities预编译中的`Multilocation`结构构建与[XCM Transactor预编译的Multilocation](/builders/interoperability/xcm/xcm-transactor#building-the-precompile-multilocation){target=\_blank}相同。
+在XCM Utilities预编译中的`Multilocation`结构构建与[XCM Transactor预编译的Multilocation](/builders/interoperability/xcm/remote-execution/substrate-calls/xcm-transactor-precompile/#building-the-precompile-multilocation){target=\_blank}相同。
 
 ## 使用XCM Utilities预编译 {: #using-the-xcmutils-precompile }
 
 XCM Utilities预编译允许用户无需前往波卡库即可通过Ethereum JSON-RPC读取数据。此功能更多的是带来了便利性，而不是为了智能合约用例。
 
-对于`multilocationToAddress`，一个示例用例是能够通过将其multilocation派生的地址加入白名单来允许来自其他平行链的交易。一个用户可以通过计算和存储地址来将multilocation加入白名单。EVM交易可以通过[远程EVM调用](/builders/interoperability/xcm/remote-evm-calls){target=\_blank}从其他平行链进行发起。
+For `multilocationToAddress`, one example use case is being able to allow transactions that originate from other parachains by whitelisting their Computed Origin addresses. A user can whitelist a multilocation by calculating and storing an address. EVM transactions can originate from other parachains via [remote EVM calls](/builders/interoperability/xcm/remote-execution/remote-evm-calls){target=\_blank}.
```
Comment (eshaben) [inline builders/interoperability/xcm/xcm-utils.md:56]: @wuzhong-papermoon

File: tutorials/interoperability/uniswapv2-swap-xcm.md [hunk 18]
```diff
@@ -32,34 +32,38 @@ _作者：Alberto Viera_
...
 |       原链解码地址       | `0xbc5f3c61709f218d983fc773a600958a07fb18047418df7eeb0501d0679e397a` |
-| 多地点衍生账户（32字节） | `0x61cd3e07fe7d7f6d4680e3e322986b7877f108ddb18ec02c2f17e82fe15f9016` |
-| 多地点衍生账户（20字节） |             `0x61cd3e07fe7d7f6d4680e3e322986b7877f108dd`             |
+| Computed Origin账户（32字节） | `0x61cd3e07fe7d7f6d4680e3e322986b7877f108ddb18ec02c2f17e82fe15f9016` |
+| Computed Origin账户（20字节） |             `0x61cd3e07fe7d7f6d4680e3e322986b7877f108dd`             |
 
-该脚本将返回32字节和20字节的地址。我们关心的是以太坊形式的账户——20字节的账户，即`0x61cd3e07fe7d7f6d4680e3e322986b7877f108dd`。请随时在[Moonscan](https://moonbase.moonscan.io/){target=\_blank}上查找您的多地点衍生帐户。接下来，您可以使用DEV Token为该帐户提供资金。
+该脚本将返回32字节和20字节的地址。我们关心的是以太坊形式的账户——20字节的账户，即`0x61cd3e07fe7d7f6d4680e3e322986b7877f108dd`。
+
+Feel free to look up your Computed Origin account on [Moonscan](https://moonbase.moonscan.io/){target=\_blank}. Next, you can fund this account with DEV tokens.
```
Comment (eshaben) [inline tutorials/interoperability/uniswapv2-swap-xcm.md:58]: @wuzhong-papermoon

**Review by eshaben — COMMENTED** [review]
@wuzhong-papermoon tagged you in everything that is still in english - can you please translate? feel free to make the changes locally on my branch and push up the changes from there - probably easier that way for some of the pages that have a lot of english left on them

**Pull request closed without merging**